### PR TITLE
Fix archive authorization - resolves issue #237

### PR DESCRIPTION
**Pull request by goneall**
Signed-off-by: Gary O'Neall <gary@sourceauditor.com>

### CODE REVIEW

**Comment by goneall** [comment]
@anshuldutt21 - pls. review

**Comment by rtgdk** [comment]
@goneall I am not sure this will fix anything. On further research, on production instance - http://13.57.134.254/app/ anybody can archive any license even without login(archive button is shown to everybody). (since the code deployed is very old) Whereas on the test instance, http://52.32.53.255/app  only for authorised people(context_dixt["authorised"] and github login is true), we show the archive/unarchive button. This was done in https://github.com/spdx/spdx-online-tools/pull/154 (which I don't think is deployed on production instance)

File: src/app/views.py [hunk 1]
```diff
@@ -1291,30 +1291,32 @@ def archiveRequests(request, license_id=None):
     returns archive_requests.html template
     """
     context_dict = {}
+    if request.user.is_authenticated():       
+        user = request.user
+        github_login = user.social_auth.get(provider='github')
+        if utils.checkPermission(user):    
+            archive = request.POST.get('archive', True)
+            license_id = request.POST.get('license_id', False)
```
Comment (rtgdk) [inline src/app/views.py:1299]: This will error out for `GET` request since request.POST will not exist, not sure though

Comment (rtgdk) [reply src/app/views.py:1299]: Also, we don't need these lines, not used anywhere
```
archive = request.POST.get('archive', True)
license_id = request.POST.get('license_id', False)
```

**Comment by goneall** [comment]
@rtgdk I'm not completely sure how this function works - I think it may be called in 2 scenerios:

- When retrieving the list of licenses using `get`
- When setting the archive flag in the database when the archive button is pushed using `post`

I'm not sure about the `ajax` flags.

I did run some tests on my local system and the code changes did fix the problem.

Without these code changes, `authorized` never gets set and `utils.checkPermission(user)` never gets called.

**Comment by rtgdk** [comment]
@goneall yeah, you are right. This function is called for both get and post request. And you right to set the `authorized` flag. But now anyone who can login to the spdx online tool - can do a post request(ajax call) to archive/unarchive the license (He won't see the archive/unarchive button on the UI but will be able to achieve it through the post request)

**Comment by rtgdk** [comment]
@goneall I have the required changes in [this commit](https://github.com/spdx/spdx-online-tools/pull/239/commits/0b72f89b24f1ac8529d9830564f8b3c5ed73cbb7). Also, regarding the github collaborator check, it currently checks for https://api.github.com/repos/spdx/tools-python/collaborators/, should we change it to spdx/license-list-XML?

**Comment by goneall** [comment]
Thanks @rtgdk - I make one more commit to change the check to work against the license-list-XML repo rather than the python tools repo.

If it looks good to you, go ahead and merge.

**Pull request merged**